src_c/surface.c

```diff
@@ -1824,31 +1824,62 @@ surf_blit(pgSurfaceObject *self, PyObject *args, PyObject *keywds)
 {
     SDL_Surface *src, *dest = pgSurface_AsSurface(self);
     SDL_Rect *src_rect, temp;
-    PyObject *argpos, *argrect = NULL;
+    PyObject *argrect = NULL;
     pgSurfaceObject *srcobject;
     int dx, dy, result;
     SDL_Rect dest_rect;
     int sx, sy;
-    int the_args = 0;
+    int the_args = 0; /* Represents kwarg: "special_flags". */
+
+    int checks = 0;
+    int loop;
 
-    static char *kwids[] = {"source", "dest", "area", "special_flags", NULL};
-    if (!PyArg_ParseTupleAndKeywords(args, keywds, "O!O|Oi", kwids,
-                                     &pgSurface_Type, &srcobject, &argpos,
-                                     &argrect, &the_args))
+    PyObject *topleft_pos = NULL;
+    PyObject *rect;
+
+    if (!PyArg_ParseTuple(args, "O!|O", &pgSurface_Type, &srcobject,
+                          &topleft_pos))
         return NULL;
 
     src = pgSurface_AsSurface(srcobject);
+
     if (!dest || !src)
         return RAISE(pgExc_SDLError, "display Surface quit");
 
-    if ((src_rect = pgRect_FromObject(argpos, &temp))) {
+    rect = pgRect_New4(0, 0, src->w, src->h);
```

Comment:
We are creating this rect each call, but we only use it when we have kwargs, so better move it inside the `else if((rect && keywds)`, and also make change the if to `else if (keywds)`
```suggestion
```